torch/fx/passes/split_module.py

```diff
@@ -260,39 +285,35 @@ def record_cross_partition_use(
             new_node.meta = node.meta.copy()
             partition.environment[node] = new_node
 
+    # original module environment dict mapping node names to nodes
+    org_mod_env: Dict[str, torch.fx.node.Node] = {}
     # Set up values to construct base module
     base_mod_env: Dict[str, torch.fx.node.Node] = {}
     base_mod_graph: torch.fx.graph.Graph = torch.fx.graph.Graph()
     base_mod_attrs: Dict[str, torch.fx.graph_module.GraphModule] = {}
-    for node in m.graph.nodes:
-        if node.op == "placeholder":
-            default_value = (
-                node.args[0] if len(node.args) > 0 else inspect.Signature.empty
-            )
-            base_mod_env[node.name] = base_mod_graph.placeholder(
-                node.target, type_expr=node.type, default_value=default_value
+    if not keep_original_order:
+        for node in m.graph.nodes:
+            base_mod_env, base_mod_attrs = construct_graph(
+                node, base_mod_env, base_mod_attrs
             )
-            base_mod_env[node.name].meta = node.meta.copy()
-        elif node.op == "get_attr":
-            base_mod_env[node.name] = base_mod_graph.get_attr(node.target)
-            base_mod_env[node.name].meta = node.meta.copy()
-            attr_val = m
-            for atom in node.target.split("."):
-                if not hasattr(attr_val, atom):
-                    raise RuntimeError(f"Node target {node.target} not found!")
-                attr_val = getattr(attr_val, atom)
-            base_mod_attrs[node.target] = attr_val
+
+    else:
+        # Go through the graph to construct the mapping dict
+        for node in m.graph.nodes:
+            if node.op == "placeholder" or "get_attr":
```

Comment:
This will always return true because the second part of the `or` condition is just a non-empty string:
```
>>> foo = "get_attr"
>>> foo == "placeholder" or "get_attr"
'get_attr'
>>> foo = "placeholder"
>>> foo == "placeholder" or "get_attr"
True
```

You want to use another == or use `in`
```
if node.op in ("placeholder", "get_attr"):
```

Reply:
thanks for pointing out!